### PR TITLE
Hacktoberfest add scalatest

### PR DESCRIPTION
**Pull request by EdgeCaseBerg**
- Removed the main class from the test folder, now instead of `test:run` to make the tests go you can just write `test`
- I'm not familiar enough with the OTF format to know exactly what the test should be testing yet, but I just took the output of the main class that I removed and made it into the expectation of the same code. That way while we go about doing other work or doing tests we can be reasonably sure that we're not adversely effecting anything.

There are probably other places to start adding tests around like the allocation and inserts of various methods but I'm not familiar enough with the code to start adding those in yet. I was thinking the ByteAllocator might be one I'll start looking into tomorrow night when I have the time to start refactoring that a bit to use `foldLeft` and etc as discussed in #2

### CODE REVIEW

File: src/test/scala/com/tsunderebug/scolortesting/OpenTypeFontTest.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+class OpenTypeFontTest extends FlatSpec {
+
+  val expected = """|
+    |4f 54 54 4f 00 01 00 10 00 01 00 00 6e 61 6d 65
+    |f2 cb aa fa 00 00 00 1c 00 00 00 4c 00 00 00 00
+    |00 00 00 03 00 2a 00 03 00 01 04 09 00 04 00 0b
+    |00 75 00 03 00 01 04 13 00 04 00 0c 00 80 00 03
+    |00 01 04 1d 00 04 00 0b 00 8c 48 65 6c 6c 6f 20
+    |57 6f 72 6c 64 48 61 6c 6c 6f 20 57 65 72 65 6c
+    |64 48 65 6a 20 56 c3 a4 72 6c 64 65 6e""".stripMargin.trim
```
Comment (TsundereBug) [inline src/test/scala/com/tsunderebug/scolortesting/OpenTypeFontTest.scala]: I think I handled my strings wrong in the sense that they should be UTF16 but I'll fix this myself.

File: build.sbt [hunk 2]
```diff
@@ -5,5 +5,6 @@ version := "0.1"
 scalaVersion := "2.12.3"
 
 libraryDependencies ++= Seq(
-  "org.typelevel" %% "spire" % "0.14.1"
-)
\ No newline at end of file
+  "org.typelevel" %% "spire" % "0.14.1",
+  "org.scalatest" %% "scalatest" % "3.0.0"
```
Comment (TsundereBug) [inline build.sbt]: Should this be in a different SBT scope?

File: src/test/scala/com/tsunderebug/scolortesting/OpenTypeFontTest.scala [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            Seq(
+              NameRecord(d = WindowsLanguage.`English`.`United States`, content = OTFString("Hello World")),
+              NameRecord(d = WindowsLanguage.`Dutch`.`Netherlands`, content = OTFString("Hallo Wereld")),
+              NameRecord(d = WindowsLanguage.`Swedish`.`Sweden`, content = OTFString("Hej Världen")),
+            )
+          )
+        )
+      ).getBytes.map((b) => "%02x".format(b.toInt)).grouped(16).map(_.mkString(" ")).mkString("\n")
+
+    assertResult(expected)(openTypeFontString)
```
Comment (TsundereBug) [inline src/test/scala/com/tsunderebug/scolortesting/OpenTypeFontTest.scala]: Is there a way to show diffs in ScalaTest?

File: src/test/scala/com/tsunderebug/scolortesting/OpenTypeFontTest.scala (hunk 1 above)
Comment (TsundereBug) [reply src/test/scala/com/tsunderebug/scolortesting/OpenTypeFontTest.scala]: I have confirmed that the strings have to be in UTF-16BE. Update this string to

```
4f 54 54 4f 00 01 00 10 00 01 00 00 6e 61 6d 65
07 bc 13 0c 00 00 00 1c 00 00 00 4c 00 00 00 00
00 00 00 03 00 2a 00 03 00 01 04 09 00 04 00 0b
00 75 00 03 00 01 04 13 00 04 00 0c 00 80 00 03
00 01 04 1d 00 04 00 0b 00 8c 00 48 00 65 00 6c
00 6c 00 6f 00 20 00 57 00 6f 00 72 00 6c 00 64
00 48 00 61 00 6c 00 6c 00 6f 00 20 00 57 00 65
00 72 00 65 00 6c 00 64 00 48 00 65 00 6a 00 20
00 56 00 e4 00 72 00 6c 00 64 00 65 00 6e
```

File: build.sbt (hunk 2 above)
Comment (EdgeCaseBerg) [reply build.sbt]: wups totally right

File: src/test/scala/com/tsunderebug/scolortesting/OpenTypeFontTest.scala (hunk 3 above)
Comment (EdgeCaseBerg) [reply src/test/scala/com/tsunderebug/scolortesting/OpenTypeFontTest.scala]: scalatest shows differences between what it expected and what it got to some extent, for example when I forget to trim the string I got:

```
[info] OpenTypeFont
[info] - should create the expected bytes *** FAILED ***
[info]   Expected "[
[info]   ]4f 54 54 4f 00 01 00...", but got "[]4f 54 54 4f 00 01 00..." (OpenTypeFontTest.scala:35)
```

Which shows that there's a newline at the start but it expected there to be nothing (indicated by the `[]`

File: src/test/scala/com/tsunderebug/scolortesting/OpenTypeFontTest.scala (hunk 1 above)
Comment (EdgeCaseBerg) [reply src/test/scala/com/tsunderebug/scolortesting/OpenTypeFontTest.scala]: I've updated the test to use this and it is now failing (as expected?)

```
[info] - should create the expected bytes *** FAILED ***
[info]   Expected "...1 00 00 6e 61 6d 65
[info]   [07 bc 13 0c 00 00 00 1c 00 00 00 4c 00 00 00 00
[info]   00 00 00 03 00 2a 00 03 00 01 04 09 00 04 00 0b
[info]   00 75 00 03 00 01 04 13 00 04 00 0c 00 80 00 03
[info]   00 01 04 1d 00 04 00 0b 00 8c 00 48 00 65 00 6c
[info]   00 6c 00 6f 00 20 00 57 00 6f 00 72 00 6c 00 64
[info]   00 48 00 61 00 6c 00 6c 00 6f 00 20 00 57 00 65
[info]   00 72 00 65 00 6c 00 64 00 48 00 65 00 6a 00 20
[info]   00 56 00 e4 00 72 00 6c 00 64 00 65 00] 6e", but got "...1 00 00 6e 61 6d 65
[info]   [f2 cb aa fa 00 00 00 1c 00 00 00 4c 00 00 00 00
[info]   00 00 00 03 00 2a 00 03 00 01 04 09 00 04 00 0b
[info]   00 75 00 03 00 01 04 13 00 04 00 0c 00 80 00 03
[info]   00 01 04 1d 00 04 00 0b 00 8c 48 65 6c 6c 6f 20
[info]   57 6f 72 6c 64 48 61 6c 6c 6f 20 57 65 72 65 6c
[info]   64 48 65 6a 20 56 c3 a4 72 6c 64 65] 6e" (OpenTypeFontTest.scala:37)
```

**Pull request merged**